documentation/content/visiting.html

```diff
@@ -6,45 +6,84 @@
+but can be set a set of targets that act like sinks for the documents -
+see <a href="../reference/vespa-cmdline-tools.html#vespa-visit-target">vespa-visit-target</a>.
 </p><p>
-Background tasks like migration, reprocessing, backup or debugging are not latency critical.
-It is more important that such tasks have little impact on the user driven load,
-and have low priority and long run time.
-Throttling ensures that only a few requests are active per content node at a time.
+Typical use cases are data dump and reprocessing that are not time sensitive.
+However, <a href="../streaming-search.html">streaming search</a>
+uses visiting in a latency-critical context.
+</p><p>
+It is possible so prioritize visit load higher/lower, and throughput can be throttled.
+</p><p>
+Performance note: Due to the bucket iteration, visiting is normally not a low-latency iteration.
+An empty cluster has 8 distribution bits and hence 256 buckets -
+dumping just one document requires iteration unless selections are used,
+and typically uses a second or two.
+To test visiting performance, use larger data sets, do not extrapolate from small.
+Use search for low latency operations on small result sets.
 </p>
 
 
 
 <h2 id="tools">Tools</h2>
 <p>
-For programmatic access, check the
-<a href="../document-api-guide.html#visitorsession">visitor session API</a>.
-</p><p>
 <a href="../reference/vespa-cmdline-tools.html#vespa-visit">vespa-visit</a> is used to run a visit operation.
 By default, vespa-visit gets visited documents and emits to stdout.
 However, the tool may specify a visitor target and be used as a tool
 to run reprocessing or migration.
 It supports keeping a progress file on disk,
 such that you can restart it if it should fail in the middle for some reason.
 </p><p>
-<em>vespa-visit-target</em> is a tool to set up an endpoint for visiting data.
+<a href="../reference/vespa-cmdline-tools.html#vespa-visit-target">vespa-visit-target</a>
+is a tool to set up an endpoint for visiting data.
 It binds to a socket or to a slobrok address,
 which is specified as a target in the visit client.
-Check <em>vespa-visit-target --help</em> for details.
+</p><p>
+For programmatic access, see the
+<a href="../document-api-guide.html#visitorsession">visitor session API</a>.
+</p>
+
+
+
+<h2 id="selection">Selection</h2>
+<p>
+To select which documents to visit, use a
+<a href="../reference/document-select-language.html">document selection expression</a>.
+</p><p>
+If co-localization of documents has been used, and the selection specifies location criteria,
+the visitor client may be able to identify one or a few buckets
+that are guaranteed to include all documents that will match the expression.
+In all other cases, visiting needs to visit all buckets in the cluster.
+</p><p>
+Note that streaming search typically relies on this behavior to ensure low latency,
+but the visitor client will only be able to map visitors to a bucket subset
+if the document selection is simple.
```

Comment:
We should consider clarifying what "simple" means in practice. The document selection itself may be arbitrarily complex as long as there is an `AND` dependency on the location.

Reply:
right. the bucket id document should be improved, so we can link to location concept there. next pr